packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo, useState } from 'react';
+import { Check } from 'react-feather';
+import uuidV4 from 'uuid/v4';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+
+import defaultClasses from './categorySort.css';
+import { mergeClasses } from '../../classify';
+import Icon from '../Icon/icon';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+
+    const [visible, setVisible] = useState(false);
+    const [sortAttribute, setSortAttribute] = useState(
+        currentSort.sortAttribute
+    );
+    const [sortDirection, setSortDirection] = useState(
+        currentSort.sortDirection
+    );
+
+    const sortElements = useMemo(() => {
+        const onChildButtonClick = sortAttribute => {
+            setSort({
+                sortAttribute: sortAttribute.attribute,
+                sortDirection: sortAttribute.sortDirection
+            });
+            setVisible(false);
+            setSortAttribute(sortAttribute.attribute);
+            setSortDirection(sortAttribute.sortDirection);
+        };
+
+        if (visible === false) {
+            return;
+        }
+
+        const sortElements = [];
+        for (const element of availableSortMethods) {
+            const key = uuidV4();
+            sortElements.push(
+                <li key={key} className={classes.child}>
+                    <button
+                        className={classes.childButton}
+                        onClick={onChildButtonClick.bind(null, element)}
+                    >
+                        <span className={classes.childText}>
+                            {element.text}
+                        </span>
+                        {element.attribute === sortAttribute &&
```

Comment:
Generally, we avoid multiline inline JSX expressions, since they defeat the readability advantages of JSX. Consider moving them out of inline expressions and into constants.

```jsx
const conditionalFoo = a && b !== "bar" && c.items
  ? <span>hello</span>
  : null

return (
  <div>
    {conditionalFoo}
  </div>
)
```